### ISSUE TITLE
Datepicker range calendar cannot be controlled with arrow keys

### ISSUE THREAD

**Issue by bakerTX** [carbon-design-system/carbon-components-react#2011]
## Environment

> Operating system
macOS Mojave 10.14.3

> Browser
Firefox 65.0.1

## Detailed description

> What version of the Carbon Design System are you using?
carbon-components-react 6.106.3

> What did you expect to happen?
After tabbing into the datepicker range inputs, I expect to navigate inside the calendar with the arrow keys. 

> What happened instead?
Instead, I found that when trying to use the arrow keys, there is some kind of focus state that appears around the calendar itself, but then the arrow keys don't do anything else.

> What WCAG 2.1 checkpoint does the issue violate?
Success Criterion 2.1.1 Keyboard: All functionality of the content is operable through a keyboard interface without requiring specific timings for individual keystrokes, except where the underlying function requires input that depends on the path of the user's movement and not just the endpoints. 
https://www.w3.org/TR/WCAG21/#keyboard

## Steps to reproduce the issue

1. Visit the Carbon-Components-React storybook
2. Navigate to the Datepicker range option
3. Use your tab key to focus onto the datepicker input
4. Attempt to use the arrow keys to navigate the popup calendar. 

I'm posting the issue here and not directly to Flatpickr because the issue doesn't arise until the Carbon implementation: the Flatpickr example handles keyboard accessibility well: (https://flatpickr.js.org/plugins/#rangeplugin-beta) I can see in the source code that Carbon adds a bit of extra functionality onto the Flatpickr dependency. I think the issue might be found there.

Note: I checked the experimental v10 component page here, too (https://next.carbondesignsystem.com/components/date-picker/code) and the issue exists there as well.

**Comment by dakahn** [comment]
The `addKeyboardEvent` method handles of focus of the calendar with a down arrow and capturing of the keyboard events for date selection. You're incorrect about this being the source of the issue.

The issue you're describing comes from the configuration option `allowInput`. There are three existing issues tracking this bug on Flatpickr's end: 

- [When allowInput=true and appendTo, left & right arrow keys prevented in the field #1688](https://github.com/flatpickr/flatpickr/issues/1688)
- [Keyboard navigation with allowInput set to true doesn't work as expected #1329](https://github.com/flatpickr/flatpickr/issues/1329)
-  [Keyboard/focus accessibility issues #1236 ](https://github.com/flatpickr/flatpickr/issues/1236)

If you disable the ability for the user to manually input the date by removing `allowInput: true` from our Flatpickr config you'll see the expected keyboard navigability.

That said I'm working on a short-term solution _on our end_ until we can get this issue resolved in Flatpickr.

**Comment by bakerTX** [comment]
Thanks for the explanation @dak! I see now the difference between these 2
modes toggled by the allowInput property. And I see now why the carbon
storybook behaves differently than the instance on the Flatpickr examples
page. Do you know, is the allowInput configuration prop exposed to
consumers of carbon-components-react? It might be a worthwhile compromise
on our end, to simply toggle the option off and regain keyboard calendar
navigation.

Anyway, It's a really interesting issue and I'm curious to see it resolved
in some way. If you want to collaborate on your next steps / short term
solution in anyway, I'd love to lend a hand.

**Comment by dakahn** [comment]
I'm going to close this out. There _are_ keyboard controls -- they're just not particularly great or documented well. We'll need to work on making the left and right arrows control the months change as well as a few other improvements, but that's been captured elsewhere. 🐬
